cmd/dagger/call.go

```diff
@@ -2,52 +2,108 @@ package main
 
 import (
 	"fmt"
+	"io"
+	"os"
+	"path/filepath"
 
 	"github.com/spf13/cobra"
 )
 
+var outputPath string
+var outputExport bool
+
 var callCmd = &FuncCommand{
 	Name:  "call",
 	Short: "Call a module function",
 	Long:  "Call a module function and print the result.\n\nOn a container, the stdout will be returned. On a directory, the list of entries, and on a file, its contents.",
+	Init: func(cmd *cobra.Command) {
+		cmd.PersistentFlags().StringVarP(&outputPath, "output", "o", "", "Path in the host to save the result to")
+	},
 	OnSelectObjectLeaf: func(c *FuncCommand, name string) error {
 		switch name {
 		case Container:
+			if outputPath != "" {
+				c.Select("export")
+				c.Arg("path", outputPath)
+				outputExport = true
+				return nil
+			}
 			// TODO: Combined `output` in the API. Querybuilder
 			// doesn't support querying sibling fields.
 			c.Select("stdout")
 		case Directory:
+			if outputPath != "" {
+				c.Select("export")
+				c.Arg("path", outputPath)
+				outputExport = true
+				return nil
+			}
 			c.Select("entries")
 		case File:
+			if outputPath != "" {
+				c.Select("export")
+				c.Arg("path", outputPath)
+				c.Arg("allowParentDirPath", true)
+				outputExport = true
+				return nil
+			}
 			c.Select("contents")
 		default:
 			return fmt.Errorf("return type %q requires a sub-command", name)
 		}
 		return nil
 	},
 	AfterResponse: func(_ *FuncCommand, cmd *cobra.Command, _ *modTypeDef, response any) error {
-		return printResponse(cmd, response)
+		writer := cmd.OutOrStdout()
+
+		if outputPath != "" {
+			path, err := filepath.Abs(outputPath)
+			if err != nil {
+				path = outputPath
```

Comment:
nit: are there any normal cases where this would error such that we want to handle it like this? On immediate thought it seems like if we can't make a path absolute something very wrong is happening

Reply:
I'm not sure when that would happen, but I need to handle the error, and this seemed the most reasonable since this is only used in the "success message", to confirm where it was saved. So it's a "best effort" thing.

For example, if you don't have permission to write to the path, I would rather that error come from `os.Create` rather than here because if there's an issue with the export/save, that'll return a better error anyway.

It's only a concern if it successfully saves the file but fails to get the absolute path to report where it was saved. In that case, falling back to the value the user provided seems reasonable.

Reply:
I moved this into a function so it only happens after successfully writing the file. Hopefully it makes it clearer to read the code.